core/src/main/scala/algebra/instances/long.scala

```diff
@@ -7,14 +7,14 @@ import algebra.ring._
 package object long extends LongInstances
 
 trait LongInstances extends cats.kernel.instances.LongInstances {
-  implicit val longAlgebra: EuclideanRing[Long] =
+  implicit val longAlgebra: LongAlgebra =
```

Comment:
why not `CommutativeRing[Long]`? Is the idea that the binary compat is more stable if we point to a single stable class?